tests/lwip_sock_ip/stack.c

```diff
@@ -197,9 +197,16 @@ void _net_init(void)
     netif_add(&netif, &netdev, lwip_netdev_init, tcpip_input);
 #endif
 #if LWIP_IPV6
-    static const uint8_t local6[] = _TEST_ADDR6_LOCAL;
+    static const uint8_t local6_a[] = _TEST_ADDR6_LOCAL;
+    /* XXX need to copy into a stack variable. Otherwise, when just using
+     * `local6_a` this leads to weird alignment problems on some platforms with
+     * netif_add_ip6_address() below */
+    ip6_addr_t local6;
```

Comment:
Should it not be this one that is `static` ? If `netif_add_ip6_address` needs a pointer that is stable in memory, it must be `static`.
The one in `local6_a` can be set as a normal variable I think.

I will try something so that it can be declared a an union of both types and still initialized using `_TEST_ADDR6_LOCAL`.

Reply:
As it still needs the `zone` init and they do not provide a static initializer in `lwip` memcpy may be the best solution. So I will not try to do an enum type here.

Reply:
@cladmi I tried already. lwIP's IPv6 addresses are just ugly. Don't put too much effort into it.

Reply:
Agreed, that is why I said "NOT try" :)

Do you have an information about the first question ? Which is in general, why do we need a `static const`.

Reply:
AFAIR `static` in stack *variables* means the variable is always assumed to be in the same place in memory. I don't think we can assume this here. `static const` means the same as with global constants: the variable is stored in ROM only, but only accessible from the block of the function (under that name).

Reply:
making it `static` wouldn't also do anything to stack usage iirc.

Reply:
`static` in a function makes the variable go into `.data` so not in the stack.
Without it will be on the stack and destroyed after the end of the function.

That is what my question is, does `netif_add_ip6_address` require a static pointer or will it copy the value ?

Reply:
It looks like it is copied so it will work.

Reply:
But then it would be better to declare the `static const` variable globally, because now it is duplicated a few times.

```
arm-none-eabi-nm bin/samr21-xpro/tests_lwip_sock_ip.elf | grep ' local'
0000ad14 t local.12505
0000ad2c t local.12510
0000ad44 t local.12515
0000ad5c t local.12524
0000ad74 t local.12529
0000ad8c t local.12542
0000ada4 t local.12551
0000adbc t local.12557
0000add4 t local.12561
0000adec t local.12567
0000ae04 t local.12574
0000ae1c t local.12582
0000ae34 t local.12588
0000ae4c t local.12595
0000ae64 t local.12602
0000ae7c t local.12611
0000ae94 t local.12616
0000aeac t local.12635
0000aec4 t local.12647
0000aedc t local.12654
0000aef4 t local.12667
0000af0c t local.12679
0000b8a0 t local6_a.13059
```